const-directives.md

```diff
@@ -2,8 +2,210 @@
+| RGBA16F | RGBA32F |                |
+
+#### Signed Integer
+
+Signed Integer buffers allow storing signed integers, just like the tin says. These are standard two's complement integers, which means they can store negative values.
+
+| 8-bit  | 16-bit  | 32-bit  |
+| ------ | ------- | ------- |
+| R8I    | R16I    | R32I    |
+| RG8I   | RG16I   | RG32I   |
+| RGB8I  | RGB16I  | RGB32I  |
+| RGBA8I | RGBA16I | RGBA32I |
+
+#### Unsigned Integer
+
+Unsigned Integer buffers allow storing of unsigned integers, which will not be interpreted as negative values (although they can technically be cast to signed integers). As such they do not directly allow the storage of negative integers, but have double the range of their signed cousins in the positive domain.
+
+| 8-bit   | 16-bit   | 32-bit   |
+| ------- | -------- | -------- |
+| R8UI    | R16UI    | R32UI    |
+| RG8UI   | RG16UI   | RG32UI   |
+| RGB8UI  | RGB16UI  | RGB32UI  |
+| RGBA8UI | RGBA16UI | RGBA32UI |
+
+
+This directive only needs to be defined once in the shader pack, can can be defined in (mostly) any shader file. Because the format names are not defined as part of glsl, the directive must either be put in a block comment, or the format name must be otherwise manually defined to any value to avoid compilation errors. Do not put the directive in a single line comment, or have any other text in the line of the directive, as this may cause it to not be detected properly.
+
+### Declaration
+
+Scope: Entire Pack
+
+```glsl
+/*
+const int <bufferName>Format = <format>;
+
+Ex:
+const int colortex4Format = RGB9_E5;
+*/
+```
+
+### Valid Declaration Locations
+
+* ❌ Vertex Shader (*.vsh)
+* ❌ Geometry Shader (*.gsh)
+* ✔️ Fragment Shader (*.fsh)
+
+### Implementation Support
+
+* ❓ ShadersMod
+* ✔️ OptiFine
+* ✔️ Iris
+
+## Texture Clear
+
+This Clear directive allows a shader pack to disable clearing for a colortex or shadowcolor color attachment. This means that the values written to the texture will be retained through the next frame instead of being cleared after every frame.
+
+The ClearColor directive is only used when clearing is enabled, and it defined the values stored in the buffer during the clear operation. By default, all colortex buffers are cleared to 0s (black), except colortex0, which is cleared to the current fog color, and colortex1, which is cleared to white. All shadowcolor buffers are cleared with white by default. This directive allows changing this clear color for each attachment.
+
+Both of these directives only need to be defined once in the shader pack, can can be defined in (mostly) any shader file.
+
+### Declaration
+
+Scope: Entire Pack
+
+```glsl
+const bool <bufferName>Clear = false;
+const vec4 <bufferName>ClearColor = <value>;
+```
+
+### Value Range
+
+* Clear
+	* Possible Values: true/false
+	* Default Value: false
+	* Out-of-range values behavior: default value (false)
+* ClearColor
+	* Possible Values: true/false
```

Comment:
Probably a typo :P

Reply:
Yup, looking more into it, doesn't look like there's a proper way to set the clear color of integer buffers. You still have to set it with a vec4, seems like for integer buffers any positive becomes the max positive, and any negative becomes the min negative, and 0 stays 0.

Reply:
Correct, this feature request would change that if someone picked it up: https://github.com/IrisShaders/Iris/issues/1167

As a workaround some shader devs have resorted to converting their int to a float representation and then writing that here. But for those who aren't aware maybe it would be useful to link that issue here